ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -408,21 +421,34 @@ function renderPodGroup(props: ApplicationResourceTreeProps, id: string, node: R
     const margin = 8;
     let topExtra = 0;
     const podGroup = node.podGroup;
+    const podGroupHealthy = (podGroup?.pods || []).filter(pod => pod.health === 'Healthy');
+    const podGroupDegraded = (podGroup?.pods || []).filter(pod => pod.health === 'Degraded');
+    const podGroupInProgress = (podGroup?.pods || []).filter(pod => pod.health === 'Progressing');
```

Comment:
We shouldn't have so many calls to filter here. Instead try something like:

```ts
const podGroupHealth = [];
const podGroupDegraded = [];
const podGroupInProgress = [];

for (const pod of podGroup?.pods || []) {
  switch (pod.health) {
    case 'Healthy':
      podGroupHealthy.push(pod);
      break;
    case 'Degraded':
      podGroupDegraded.push(pod);
      break;
    case 'Progressing':
      podGroupInProgress.push(pod);
  }
}
```

This way, we only loop through the list once.